homeassistant/components/hydrawise/config_flow.py

```diff
@@ -21,26 +23,62 @@
 class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
     """Handle a config flow for Hydrawise."""
 
-    VERSION = 1
+    VERSION = 2
 
-    async def _create_entry(
-        self, api_key: str, *, on_failure: Callable[[str], FlowResult]
+    def __init__(self) -> None:
+        """Construct a ConfigFlow."""
+        self.reauth_entry: ConfigEntry | None = None
+
+    async def _create_or_update_entry(
+        self,
+        api_key: str,
+        username: str,
+        password: str,
+        *,
+        on_failure: Callable[[str], FlowResult],
     ) -> FlowResult:
         """Create the config entry."""
-        api = legacy.LegacyHydrawiseAsync(api_key)
+
+        # Verify that the provided credentials work."""
+        if api_key != "":
+            api = legacy.LegacyHydrawiseAsync(api_key)
+        else:
+            api = client.Hydrawise(client.Auth(username, password))
```

Comment:
Again here, use `Auth` from the `auth` module where it's defined.

Reply:
Fixed.